src/components/suggest/suggest_input.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { EuiFilterButton } from '../filter_group';
+import { EuiFieldText } from '../form';
+import { EuiToolTip } from '../tool_tip';
+import { EuiIcon } from '../icon';
+import { EuiPopover, EuiInputPopover } from '../popover';
+
+export class EuiSuggestInput extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      isPopoverOpen: false,
+    };
+  }
+
+  onFieldChange(e) {
+    this.setState({
+      value: e.target.value,
+      isPopoverOpen: e.target.value !== '' ? true : false,
+    });
+    this.props.sendValue(e.target.value);
+  }
+
+  closePopover() {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  }
+
+  render() {
+    const {
+      className,
+      status,
+      append,
+      prefix,
+      tooltipContent,
+      suggestions,
+      sendValue,
+      ...rest
+    } = this.props;
+
+    const statusMap = {
+      unsaved: {
+        icon: 'dot',
+        color: 'accent',
+      },
+      saved: {
+        icon: 'checkInCircleFilled',
+        color: 'secondary',
+      },
+      unchanged: {
+        icon: '',
+        color: 'secondary',
+      },
+    };
+
+    let icon;
+    let color;
+    let tooltip;
+
+    if (statusMap[status]) {
+      icon = statusMap[status].icon;
+      color = statusMap[status].color;
+    }
+    const classes = classNames('euiSuggestInput', className);
+
+    // EuiFieldText's append accepts an array of elements so start by creating an empty array
+    const appendArray = [];
+
+    const statusElement = status !== 'isLoading' && (
+      <EuiToolTip
+        position="left"
+        content={tooltipContent}
```

Comment:
I think you can still supply some good defaults in the component itself for the tooltip instead of forcing that on the consumer. Probably just add a few to the `statusMap` but allow `tooltipContent` to override.
```suggestion
        content={tooltipContent || statusMap[status].tooltip}
```

The `||` will evaluate the first option first and if it is false, move to the next option.